seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/sql/zeta/functions/SystemFunction.java

```diff
@@ -70,11 +68,13 @@ public static Object nullif(List<Object> args) {
         return v1;
     }
 
-    public static Object array(List<Object> args) {
+    public static String[] array(List<Object> args) {
```

Comment:
Why does it return a fixed string[]?

Example:
Array(1, 2, 3) returns Integer[]
Array(1L, 2, 3) returns Long[]
Array(1L, 1.1F, 2.1D) returns Double[]
Array(true, false, true) returns Boolean[]
Array(true, "aaa", 1) returns String[]

Reply:
case ZetaSQLFunction.ARRAY:
       return ArrayType.STRING_ARRAY_TYPE;

Then we need to dynamically generate the returned ArrayType.

Reply:
> Then we need to dynamically generate the returned ArrayType.

+1, we can reuse https://github.com/apache/seatunnel/blob/d2576ff13112b0f845930374632a959cf29e7eb9/seatunnel-api/src/main/java/org/apache/seatunnel/api/table/type/TypeUtil.java#L23 to check and convert type.